pyproject.toml

```diff
@@ -15,13 +15,13 @@ astor = "^0.8.1"
 openai = "<2"
 ipython = "^8.13.1"
 matplotlib = "^3.7.1"
-pydantic = "^1"
+pydantic = ">=2.5.2"
```

Comment:
I see you changed the pydantic version from 1 to >2.0 . And I think thats a ground breaking change. Because in Version 2.0 alot of Pydantic 1 changes got deprecated(or the way to use them changed). So I do have doubt by changing the Version to >2.0 will break something.

Reply:
@Zidane786 thanks for your review. But mistralai is not working if pydantic is < 2.5.2. Can you suggest me some way for this issue.

Reply:
@gventuri could you also once look into this?